filters/flowid/generator-ulid.go

```diff
@@ -0,0 +1,40 @@
+package flowid
+
+import (
+	"github.com/oklog/ulid"
+	"io"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type ulidGenerator struct {
+	sync.Mutex
+	r io.Reader
+}
+
+func newULIDGenerator() flowIDGenerator {
+	return newULIDGeneratorWithEntropyProvider(rand.New(rand.NewSource(time.Now().UTC().UnixNano())))
+}
+
+func newULIDGeneratorWithEntropyProvider(r io.Reader) flowIDGenerator {
+	return &ulidGenerator{r: r}
+}
+
+func (g *ulidGenerator) Generate() (string, error) {
+	g.Lock()
```

Comment:
Parallel benchmarks:

New entropy provider on each request:
```
BenchmarkFlowIdULIDGenerator-8   	 1000000	      2543 ns/op	    5456 B/op	       4 allocs/op
```

Using the mutex:
```
BenchmarkFlowIdULIDGenerator-8   	 5000000	       384 ns/op	      48 B/op	       2 allocs/op
```